web/js/layoutManager.js

```diff
@@ -145,10 +145,10 @@ LayoutViewport, ItemsHandler */
     if (!currentLayout || !isOnGoing(CandidateLayout)) {
       currentLayout && currentLayout.destroy();
       currentLayout = new CandidateLayout(container, items, item);
-      Utils.sendEvent('layoutManager:layoutChanged');
     }
 
     currentLayout.rearrange();
+    Utils.sendEvent('layoutManager:layoutChanged');
```

Comment:
This event isn't sent for all layout changes. For instance, double-click a non-screen-sharing stream view. The `itemSelected` handler (in line 26) of layoutManager.js is called, but it doesn't call the `rearrange()` function.

Reply:
And if you double-click the screen-sharing stream view after double-clicking another view, the size of the canvas is wrong.

Reply:
And I noticed that if you join a room while a screen-sharing stream already exists, the screen-sharing stream view does not get the big layout treatment. It should. (That big layout for the screen-sharing stream view only happens when a screen-sharing stream is started while you are already in the room.) It's outside of the scope of this ticket, but we should fix that too.